packages/yaml/src/index.ts

```diff
@@ -0,0 +1,146 @@
+import { type Service } from '@volar/language-service';
+import { type TextDocument } from 'vscode-languageserver-textdocument';
+import { type LanguageSettings, type LanguageService } from 'yaml-language-server';
+import { getLanguageService } from 'yaml-language-server/lib/umd/languageservice/yamlLanguageService.js';
+
+export interface Provide {
+	'yaml/languageService': () => LanguageService;
+}
+
+function isYaml(document: TextDocument): boolean {
+	return document.languageId === 'yaml';
+}
+
+function noop(): undefined { }
+
+/**
+ * Create a Volar language service for YAML documents.
+ */
+export default (settings: LanguageSettings): Service<Provide> => {
+	return (context): ReturnType<Service<Provide>> => {
+
+		if (!context) {
+			return { triggerCharacters: [' ', ':'] } as any;
+		}
+
+		const ls = getLanguageService({
+			async schemaRequestService(uri) {
+				if (uri.startsWith('file:') && context?.env.fs) {
+					const result = await context?.env.fs.readFile(uri);
+					if (result) {
+						return result;
+					}
+
+					throw new Error(`No such file: ${uri}`);
+				}
+
+				// @ts-expect-error This exists as an experimental API in Node 16.
+				const response = await fetch(uri);
```

Comment:
We don't need to handle the http request in the service because it is already handled by the language server: https://github.com/volarjs/volar.js/blob/fc0bc2c262e693b6d41e87bff758ac1eb6eeb8c9/packages/language-server/src/node/index.ts#L54